libs/domain/push-notification/src/services/default-push.service.ts

```diff
@@ -0,0 +1,34 @@
+import { HttpService } from '@spryker-oryx/core';
+import { inject } from '@spryker-oryx/di';
+import { Observable, switchMap } from 'rxjs';
+import { PushProvider } from './providers';
+import { pushApiType, PushService, PushServiceConfig } from './push.service';
+
+export class DefaultPushService implements PushService {
+  constructor(
+    protected provider = inject(PushProvider),
+    protected config = inject(PushServiceConfig),
+    protected http = inject(HttpService)
+  ) {}
+
+  init(): Observable<void> {
+    return this.provider.init();
+  }
+
+  subscribe(): Observable<any> {
+    return this.provider
+      .getSubscription()
+      .pipe(switchMap((token) => this.sendToken(token)));
+  }
+
+  unsubscribe(): Observable<boolean> {
+    return this.provider.deleteSubscription();
+  }
+
+  protected sendToken(token: any): Observable<any> {
```

Comment:
Can we specify a type here?

Reply:
It depends on the provider. Firebase returns a different type than web-push. And it only seems possible to define the provider type (`TSubscription`) when calling the constructor unless there's a nicer way to do it.

And this service should probably be agnostic to the provider type anyways.